src/main/java/uk/gov/hmcts/reform/sscs/config/CitizenCcdService.java

```diff
@@ -105,6 +131,55 @@ private CaseDetails newCase(SscsCaseData caseData, String eventType, String summ
         return caseDetails;
     }
 
+    @Retryable
+    public CaseDetails triggerCaseEventV2(String caseId, String eventType, String summary, String description, IdamTokens idamTokens) {
+        return updateCaseV2(caseId, eventType, idamTokens, data -> new UpdateResult(summary, description));
+    }
+
+    public record UpdateResult(String summary, String description) { }
+
+    /**
+     * Update a case while making correct use of CCD's optimistic locking.
+     * Changes can be made to case data by the provided consumer which will always be provided
+     * the current version of case data from CCD's start event.
+     */
+    @Retryable
+    public CaseDetails updateCaseV2(String caseId, String eventType, IdamTokens idamTokens, Function<SscsCaseData, UpdateResult> mutator) {
```

Comment:
can this go into common's ? If it is used in multiple repos may be worth doing else we can keep it here for now

Reply:
@nitinprabhuhmcts from a search of the codebase I couldn't see any references to the update case method being replaced in CitizenCcdService outside of this repo. The rest used the existing caseworker update function